src/Traits/LaratrustHasAssignedPermissionsTrait.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+namespace Laratrust\Traits;
+
+/**
+ * This file is part of Laratrust,
+ * a role & permission management solution for Laravel.
+ *
+ * @license MIT
+ * @package Laratrust
+ */
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Config;
+
+
+trait LaratrustHasAssignedPermissionsTrait
+{
+	use LaratrustHasEvents;
+
+	/**
+	 * Boots the role model and attaches event listener to
+	 * remove the many-to-many records when trying to delete.
+	 * Will NOT delete any records if the role model uses soft deletes.
+	 *
+	 * @return void|bool
+	 */
+	public static function bootLaratrustHasAssignedPermissionsTrait()
+	{
+		$flushCache = function ($model) {
+			$model->flushCache();
+		};
+
+		// If the model doesn't use SoftDeletes.
+		if (method_exists(static::class, 'restored')) {
+			static::restored($flushCache);
+		}
+
+		static::deleted($flushCache);
+		static::saved($flushCache);
+
+		static::deleting(function ($model) {
+			if (method_exists($model, 'bootSoftDeletes') && !$model->forceDeleting) {
+				return;
+			}
+			$model->syncPermissions();
+		});
+	}
+	/*
+	 *
+	 * - - -  R E L A T I O N S H I P S  - - -
+	 *
+	 */
+	/**
+	 * Get the Permissions for the .
+	 */
+	public function permissions()
+	{
+		return $this->morphToMany(Config::get('laratrust.models.permission'), 'model',
+			Config::get('laratrust.tables.permission_models'));
+	}
+
+
+	/*
+	 *
+	 * - - -   M E T H O D S  - - -
+	 *
+	 */
+
+	/**
+	 * Tries to return all the cached permissions of the model.
+	 * If it can't bring the permissions from the cache,
+	 * it brings them back from the DB.
+	 *
+	 * @return \Illuminate\Database\Eloquent\Collection
+	 */
+	public function cachedPermissions()
+	{
+		if (!Config::get('laratrust.use_cache')) {
+			return $this->permissions()->get();
+		}
+		$cacheKey = $this->getCacheKey();
+
+		return Cache::remember($cacheKey, Config::get('cache.ttl', 60), function () {
+			return $this->permissions()->get()->toArray();
```

Comment:
This is because if we don't use the cache we don't have the necessity to hydrate the model again, because we already have it. And when it hits the cache, the idea is to store it as an array, so we don't have a huge amount of data in the cache store only for one user.

This is the only way I found to avoid storing the serialized model into the cache. If you think of another solution for this or how to improve it, it would be great.